cmake/app/boilerplate.cmake

```diff
@@ -313,3 +313,6 @@ foreach(boilerplate_lib ${ZEPHYR_INTERFACE_LIBS_PROPERTY})
     ${boilerplate_lib}
     )
 endforeach()
+
+target_link_libraries_ifdef(CONFIG_BT      app subsys_bluetooth)
+target_link_libraries_ifdef(CONFIG_NET_BUF app subsys_net)
```

Comment:
Core build script code such as boilerplate.cmake must be agnostic to the details of subsystems.

For this use-case, one should take advantage of the CONFIG_APP_LINK_WITH_${boilerplate_lib_upper_case}
feature. See above foreach loop.

Reply:
yes, as I stated in the commit message:

> This is currently a temporary solution to ensure all samples and tests
> builds correctly.
> 
> As part of the RFC it should be decided what the optimal approach for
> linking should be.
> 

So the loop is a candidate.
Another approach could be to target_link_libraries where the subsystem is actually included, e.g.
as part of:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/CMakeLists.txt#L1-15

So when a subsystem is included, it is also linked to app.